tests/test_cli.py

```diff
@@ -375,12 +430,11 @@ def test_run_non_ascii_file(self):
 
         # Make sure the default localization conditions on this "system"
         # support UTF-8 encoding.
-        loc = locale.getlocale()
         try:
-            locale.setlocale(locale.LC_ALL, 'C.UTF-8')
+            locale.setlocale(locale.LC_ALL, (None, 'UTF-8'))
         except locale.Error:
-            locale.setlocale(locale.LC_ALL, 'en_US.UTF-8')
-        self.addCleanup(locale.setlocale, locale.LC_ALL, loc)
+            self.skipTest('no UTF-8 locale available')
+        self.addCleanup(locale.setlocale, locale.LC_ALL, (None, None))
```

Comment:
I now understand more how the whole `locale` module works, so I improved this test case (unrelated to my changes) as well. This should now work for any available `UTF-8` locale, I guess - and also reset to the "default" from before better then with `getlocale`.

Reply:
Interesting: `'C.UTF-8'` → `(None, 'UTF-8')`

Thanks!